src/Infrastructure.VS.UnitTests/VsInfoServiceTests.cs

```diff
@@ -54,22 +67,34 @@ public void Create_VsShellCallSucceeds_ReturnsExpectedPath()
         public void Create_VsShellCallFails_ExceptionThrown()
         {
             var logger = new TestLogger();
-            var serviceProvider = CreateConfiguredServiceProvider("c:\\test\\", shellHrResult: -123);
+            var serviceOp = CreateConfiguredServiceOperation("c:\\test\\", shellHrResult: -123);
 
-            Action act = () => new VsInfoService(serviceProvider.Object);
+            var testSubject = new VsInfoService(serviceOp);
+
+            Action act = () => _ = testSubject.InstallRootDir;
             act.Should().ThrowExactly<COMException>();
         }
 
-        private Mock<IServiceProvider> CreateConfiguredServiceProvider(string installDirectory, int shellHrResult = VSConstants.S_OK)
```

Comment:
Fix up the existing tests to use the new service.
I've also manually debugged through the code in VS to check the threading works as expected.